docker/docker_image_src.go

```diff
@@ -798,7 +798,7 @@ func (n *bufferedNetworkReader) read(p []byte) (int, error) {
 	select {
 	case b = <-n.readyBuffer:
 		if b.err != nil {
-			if b.err != io.EOF {
+			if b.err != io.EOF { //nolint:errorlint
```

Comment:
Returning _exactly_ `io.EOF` is a clearly documented API, so this just indicates an impractical linter to me.

Reply:
errorlint has a list of exceptions for that, listing functions that can return literal `io.EOF` but the mechanism doesn't work for some of the more complex setups (including, apparently, this one, when an error is part of a struct which is passed through a channel).